app/utils/AccountUtils.scala

```diff
@@ -17,18 +17,23 @@
 package utils
 
 import controllers.auth.AuthenticatedRequest
-import models.ActingAsAttorneyFor
+import models.{ActingAsAttorneyFor, AgentToken}
 import play.api.i18n.Messages
 import play.api.mvc.Request
 import uk.gov.hmrc.domain.{SaUtr, TaxIdentifier, Uar}
 
 trait AccountUtils {
-  def getAccount(request: AuthenticatedRequest[_]): TaxIdentifier =
-    request.agentRef.getOrElse(request.saUtr.getOrElse(SaUtr("")))
+  def getAccount(request: AuthenticatedRequest[_]): Option[TaxIdentifier] =
+    request match {
+      case x if (x.agentRef.isDefined) => Some(x.agentRef.get)
```

Comment:
x is request so you are not really matching on anything here, can you see a nicer way of writing this?